### PR TITLE
docs: explain why typescript is added either way

### LINKED ISSUE #7475: Error while creating 'Library skeleton project'
### Describe the bug

I have just started a new project using the `npm create svelte@latest` command. After which I chose the following options:

```
✔ Where should we create your project?
  (leave blank to use current directory) … 
✔ Which Svelte app template? › Library skeleton project
✔ Add type checking with TypeScript? › No
✔ Add ESLint for code linting? … No / Yes
✔ Add Prettier for code formatting? … No / Yes
✔ Add Playwright for browser testing? … No / Yes
```

Then run `npm install`  and upon completion `npm run build`. I received the following error:

```
> Failed to locate tsconfig or jsconfig
    at loadTsconfig (/Users/stuymedova/Documents/github/projects/web-development/_test2/node_modules/svelte2tsx/index.js:7260:15)
    at emitDts (/Users/stuymedova/Documents/github/projects/web-development/_test2/node_modules/svelte2tsx/index.js:7250:36)
    at async emit_dts (file:///Users/stuymedova/Documents/github/projects/web-development/_test2/node_modules/@sveltejs/package/src/typescript.js:23:2)
    at async Module.build (file:///Users/stuymedova/Documents/github/projects/web-development/_test2/node_modules/@sveltejs/package/src/index.js:30:3)
    at async file:///Users/stuymedova/Documents/github/projects/web-development/_test2/node_modules/@sveltejs/package/src/cli.js:27:4
```

### Reproduction

I suppose that instructions above would be sufficient because I've started from scratch.

### Logs

_No response_

### System Info

```Shell
System:
    OS: macOS 13.0
    CPU: (4) x64 Intel(R) Core(TM) i3-1000NG4 CPU @ 1.10GHz
    Memory: 28.30 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 8.11.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 107.0.5304.87
    Firefox: 98.0
    Safari: 16.1
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.86 
    @sveltejs/kit: next => 1.0.0-next.532 
    @sveltejs/package: next => 1.0.0-next.1 
    svelte: ^3.44.0 => 3.52.0 
    vite: ^3.1.0 => 3.2.2
```

### Severity

blocking an upgrade

### Additional Information

_No response_

### LINKED ISSUE #8189: TypeScript added when not requested
### Describe the bug

When I create a new SvelteKit Library project using `npm create svelte@latest <name>`, I specifically request to have no TypeScript, but the package.json file has typescript and tslib as dev dependencies. Very annoying to have to remove them.

### Reproduction

create a new SvelteKit library project, requesting no typescript.

### Logs

```Shell
$ npm create svelte@latest temp

create-svelte version 2.0.1

Welcome to SvelteKit!

√ Which Svelte app template? » Library skeleton project
√ Add type checking with TypeScript? » No
√ Add ESLint for code linting? ... No / Yes
√ Add Prettier for code formatting? ... No / Yes
√ Add Playwright for browser testing? ... No / Yes
√ Add Vitest for unit testing? ... No / Yes

Your project is ready!

Install community-maintained integrations:
  https://github.com/svelte-add/svelte-adders

Next steps:
  1: cd temp
  2: npm install (or pnpm install, etc)
  3: git init && git add -A && git commit -m "Initial commit" (optional)
  4: npm run dev -- --open

To close the dev server, hit Ctrl-C

Stuck? Visit us at https://svelte.dev/chat
```

### System Info

```Shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
    Memory: 8.60 GB / 15.77 GB
  Binaries:
    Node: 18.3.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.2.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.963.0), Chromium (108.0.1462.46)
    Internet Explorer: 11.0.22621.1
```

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by dummdidumm**
for a library project
closes #8189
also always add jsconfig.json, closes #7475

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: documentation/docs/30-advanced/70-packaging.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ A component library has the exact same structure as a SvelteKit app, except that
...
 
 - All the files in `src/lib`, unless you [configure](/docs/configuration) custom `include`/`exclude` options. Svelte components will be preprocessed, TypeScript files will be transpiled to JavaScript.
-- Type definitions (`d.ts` files) which are generated for Svelte, JavaScript and TypeScript files. You need to install `typescript >= 4.0.0` for this. Type definitions are placed next to their implementation, hand-written `d.ts` files are copied over as is. You can [disable generation](/docs/configuration), but we strongly recommend against it.
+- Type definitions (`d.ts` files) which are generated for Svelte, JavaScript and TypeScript files. You need to install `typescript >= 4.0.0` for this. Type definitions are placed next to their implementation, hand-written `d.ts` files are copied over as is. You can [disable generation](/docs/configuration), but we strongly recommend against it: People using your library might use TypeScript, for which they require these type definition files.
```
Comment (Rich-Harris) [inline documentation/docs/30-advanced/70-packaging.md:16]:
```suggestion
- Type definitions (`d.ts` files) which are generated for Svelte, JavaScript and TypeScript files. You need to install `typescript >= 4.0.0` for this. Type definitions are placed next to their implementation, hand-written `d.ts` files are copied over as is. You can [disable generation](/docs/configuration), but we strongly recommend against it — people using your library might use TypeScript, for which they require these type definition files.
```

**Review by Rich-Harris — APPROVED** [review]

File: packages/create-svelte/bin.js [hunk 2]
```diff
@@ -136,6 +136,10 @@ async function main() {
 	} else if (options.types === 'checkjs') {
 		console.log(bold('✔ Type-checked JavaScript'));
 		console.log('  https://www.typescriptlang.org/tsconfig#checkJs');
+	} else if (options.template === 'skeletonlib') {
+		console.log(
+			'🛈 You chose to not add type checking, but TypeScript will still be installed in order to generate type definitions when building the library'
```
Comment (Rich-Harris) [inline packages/create-svelte/bin.js:141]: what character is this supposed to be? it shows up empty in my terminal

Comment (dummdidumm) [reply packages/create-svelte/bin.js:141]: Huh, I saw it locally - it's supposed to be an info sign

Comment (dummdidumm) [reply packages/create-svelte/bin.js]: I replaced it with a warning sign, which looks better in combination with the check marks anyway - hopefully you see that one now

Comment (Rich-Harris) [reply packages/create-svelte/bin.js]: i do, though the ⚠ character tends to look awful in a terminal context — on this MBP, where excellent font rendering is something you take for granted, it's a pixelated mess. I swapped it out for a yellow ▲ character instead

**Review by Rich-Harris — APPROVED** [review]

**Comment by idleman** [comment]
"You chose to not add type checking, but TypeScript will still be installed in order to generate type definitions when building the library". Must projects never do this. so no, it is not a reason at all.

**Comment by Rich-Harris** [comment]
> Must projects never do this. so no, it is not a reason at all.

This change only applies if you create a library project. If you're creating a library project then yeah, you're gonna build it

**Comment by idleman** [comment]
> > Must projects never do this. so no, it is not a reason at all.
> 
> This change only applies if you create a library project. If you're creating a library project then yeah, you're gonna build it

Thanks for clarification. Thought it was always installed. You never know these days.

**Pull request merged**